client/src/document.js

```diff
@@ -133,47 +133,73 @@ function SidebarLeaf({ depth, title, content }) {
   );
 }
 
-function RenderHTMLElementDocument({ document }) {
-  let sections = [];
-
-  sections.push(
-    <Prose key="short_description" section={document.prose.short_description} />
-  );
-  sections.push(
-    <InteractiveExample key="interactive_example" document={document} />
-  );
-  sections.push(<Prose key="overview" section={document.prose.overview} />);
-  sections.push(<Attributes key="attributes" document={document} />);
-  sections.push(
-    <ProseWithHeading key="usage_notes" section={document.prose.usage_notes} />
-  );
-  sections.push(
-    <ProseWithHeading
-      key="accessibility_concerns"
-      section={document.prose.accessibility_concerns}
-    />
-  );
-  sections.push(<Examples key="examples" document={document} />);
-  sections.push(
-    <BrowserCompatibility key="browser_compatibility" document={document} />
-  );
-  sections.push(
-    <ProseWithHeading key="see_also" section={document.prose.see_also} />
-  );
+function RenderDocumentBody({ document }) {
+  const sections = [];
+  /**
+   * The reason we can't return a filtered map of 'document.body' is because
+   * some of the parts of 'document.body' is an array.
+   * E.g. document.body.additional_prose == [
+   *     {type: 'prose', value: STUFF},
+   *     {type: 'prose', value: OTHER_STUFF},
+   * ]
+   */
+  document.body.forEach((section, i) => {
+    if (section.type === "prose") {
```

Comment:
Prose sections (**aside from additional_prose**, see below) do not include the H2 heading in `value.content`. It is stored separately in `value.title`. This is intentional, because in some contexts a renderer wants to include the title as a heading and in some it does not.

For example, suppose someone wants to build an 'accessibility inspector' using stumptown-content. This tool might display the content of `accessibility_concerns` when the user selects a particular tag. This probably doesn't want to display the title "Accessibility concerns" here, because the context is obvious given the tool. But MDN does want to display the title "Accessibility concerns" when it renders a complete page about that tag.

So, whether to render the title is a rendering decision, and the stumptown-content JSON is supposed to be agnostic about it.

And an MDN page will sometimes want to render H2 headings, and sometimes won't. Specifically, we don't want to render H2 headings for `short_description` and `overview`. So the renderer includes two different components, `ProseWithHeading` and `Prose`, where the first one renders an H2 from `value.title`, and the first doesn't. The renderer should choose `Prose` for `short_description` and `overview`, and `ProseWithHeading` for the rest.

At the moment, this PR chooses `Prose` for all named prose sections, with the result that all H2 headings are missing in the rendered output.

I think you might now ask that the JSON output should include a hint to the renderer about whether it should render headings for a particular section or not, so that the renderer can be generic and not special-case `short_description` and `overview`. I would feel uncomfortable with that, as it seems to be embedding rendering choices in stumptown-content. But I'm happy to open this up to debate :).